src/core/lv_refr.c

```diff
@@ -1078,7 +1078,10 @@ static void wait_for_flushing(lv_display_t * disp)
     lv_display_send_event(disp, LV_EVENT_FLUSH_WAIT_START, NULL);
 
     if(disp->flush_wait_cb) {
-        disp->flush_wait_cb(disp);
+        if(disp->flushing) {
+            disp->flush_wait_cb(disp);
+        }
+        disp->flushing = 0;
```

Comment:
```c
if(disp->flushing && disp->flush_wait_cb) {
    disp->flush_wait_cb(disp);
    disp->flushing = 0;
}
```

Reply:
I think the original code conveys what's the intention better. By looking in to a larger snippet:
```c
    if(disp->flush_wait_cb) { //Do this if the flush_wait_cb is set
        if(disp->flushing) {
            disp->flush_wait_cb(disp);
        }
        disp->flushing = 0;
    }
    else { //This is without flush_wait_cb
        while(disp->flushing);
    }
```

Reply:
Sorry, you are right. I missed the case of `else`.